video/out/w32_common.c

```diff
@@ -1563,42 +1556,32 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
             reinit_window_state(w32);
         return VO_TRUE;
     case VOCTRL_ONTOP:
-        reinit_window_state(w32);
+        update_window_state(w32);
         return VO_TRUE;
     case VOCTRL_BORDER:
-        reinit_window_state(w32);
+        update_window_style(w32);
+        update_window_state(w32);
         return VO_TRUE;
     case VOCTRL_GET_FULLSCREEN:
         *(bool *)arg = w32->current_fs;
         return VO_TRUE;
-    case VOCTRL_GET_UNFS_WINDOW_SIZE: {
-        int *s = arg;
-
-        if (!w32->window_bounds_initialized)
-            return VO_FALSE;
-
-        s[0] = w32->current_fs ? w32->prev_width : w32->dw;
-        s[1] = w32->current_fs ? w32->prev_height : w32->dh;
-        return VO_TRUE;
-    }
+    case VOCTRL_GET_UNFS_WINDOW_SIZE:
```

Comment:
Not sure it's best to collapse the VOCTRL_GET_UNFS_WINDOW_SIZE and VOCTRL_SET_UNFS_WINDOW_SIZE cases. There's only a couple of lines of shared logic, so it's probably clearer to keep them separate.